src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -334,7 +334,7 @@ private static string GetTagName(Block tagBlock)
             return textSymbol.Type == HtmlSymbolType.WhiteSpace ? null : textSymbol.Content;
         }
 
-        private static bool IsSelfClosing(Block beginTagBlock)
+        internal static bool IsSelfClosing(Block beginTagBlock)
```

Comment:
Move this method into the `TagHelperBlockRewriter` and make it `private` again. The one call to this method in this file can just pull the `SelfClosing` value off of the builder.